lib/charms/vault_k8s/v0/vault_client.py

```diff
@@ -102,6 +101,17 @@ def set_token(self, token: str) -> None:
         """Set the Vault token for authentication."""
         self._client.token = token
 
+    def approle_login(self, role_id: str, secret_id: str) -> None:
+        """Authenticate to Vault using AppRole."""
+        self._client.auth.approle.login(role_id=role_id, secret_id=secret_id, use_token=True)
+
+    def configure_charm_access_policy(self, name: str) -> None:
+        """Create/update a policy within vault for the charm to access vault."""
+        with open("src/templates/charm_policy.hcl", "r") as f:
```

Comment:
Should this file path be a parameter? This is a lib now and as it is, users of this lib are forced to have their charm policy file with this specific name in this specific path.

Reply:
Understood, as long as this is tracked and we don't forget it I'm ok

Reply:
I've noted this as something to follow up on if it isn't tackled during the refactor